Yosemite/Yosemite/Actions/ProductCategoryAction.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+import Networking
+
+/// ProductCategoryAction: Defines all of the Actions supported by the ProductCategoryStore.
+///
+public enum ProductCategoryAction: Action {
+
+    /// Retrieve ProductCategories matching the specified criteria.
+    ///
+    case retrieveProductCategories(siteID: Int64, pageNumber: Int, pageSize: Int, onCompletion: ([ProductCategory]?, Error?) -> Void)
```

Comment:
We should probably try to use the same names for the action and the `ProductCategoryStore` method. 

```suggestion
    case synchronizeProductCategories(siteID: Int64, pageNumber: Int, pageSize: Int, onCompletion: ([ProductCategory]?, Error?) -> Void)
```

Reply:
I left it as `retrieveProductCategories` because this doesn't synchronize them yet. But since synchronization PR is coming it doesn't hurt to change it now!